Realm/ObjectServerTests/SwiftCollectionSyncTests.swift

```diff
@@ -0,0 +1,209 @@
+            let realm = try openRealm(partitionValue: #function, user: user)
+            if isParent {
+                try realm.write {
+                    realm.deleteAll()
+                }
+                waitForDownloads(for: realm)
+                checkCount(expected: 0, realm, SwiftCollectionSyncObject.self)
+                executeChild()
+                waitForDownloads(for: realm)
+                checkCount(expected: 1, realm, SwiftCollectionSyncObject.self)
+                // Run the child again to add the values
+                executeChild()
+                waitForDownloads(for: realm)
+                checkCount(expected: 1, realm, SwiftCollectionSyncObject.self)
+                let object = realm.objects(SwiftCollectionSyncObject.self).first!
+                XCTAssertEqual(object[keyPath: keyPath].count, values.count*2)
+                for (el, ex) in zip(object[keyPath: keyPath], values + values) {
+                    XCTAssert(el == ex, "\(el) is not equal to \(ex)")
+                }
+                // Run the child again to delete the last 3 objects
+                executeChild()
+                waitForDownloads(for: realm)
+                XCTAssertEqual(object[keyPath: keyPath].count, values.count)
+                // Run the child again to modify the first element
+                executeChild()
+                waitForDownloads(for: realm)
+                XCTAssertEqual(object[keyPath: keyPath][0], values[1])
+            } else {
+                guard let object = realm.objects(SwiftCollectionSyncObject.self).first else {
+                    try realm.write({
+                        realm.add(SwiftCollectionSyncObject())
+                    })
+                    waitForUploads(for: realm)
+                    checkCount(expected: 1, realm, SwiftCollectionSyncObject.self)
+                    return
+                }
+
+                if object[keyPath: keyPath].count == 0 {
+                    try realm.write({
+                        object[keyPath: keyPath].append(objectsIn: values + values)
+                    })
+                    XCTAssertEqual(object[keyPath: keyPath].count, values.count*2)
+                } else if object[keyPath: keyPath].count == 6 {
+                    try realm.write({
+                        object[keyPath: keyPath].removeSubrange(3...5)
+                    })
+                    XCTAssertEqual(object[keyPath: keyPath].count, values.count)
+                } else {
+                    try realm.write({
+                        object[keyPath: keyPath][0] = values[1]
+                    })
+                    XCTAssertEqual(object[keyPath: keyPath][0], values[1])
+                }
+                waitForUploads(for: realm)
+                checkCount(expected: 1, realm, SwiftCollectionSyncObject.self)
+            }
+        } catch {
+            XCTFail("Got an error: \(error) (process: \(isParent ? "parent" : "child"))")
```

Comment:
Is there any advantage to not just having this method throw? Too much indentation makes this slightly harder to read.

Reply:
Resolved in https://github.com/realm/realm-cocoa/pull/7190